src/common/transformations/src/transformations/mlir/convert.cpp

```diff
@@ -279,9 +279,6 @@ bool elementwise_f32_binary_no_broadcast_predicate(const ov::Output<ov::Node>& o
     auto input_shape_a = output.get_node_shared_ptr()->get_input_partial_shape(0);
     auto input_shape_b = output.get_node_shared_ptr()->get_input_partial_shape(1);
     auto output_shape = output.get_partial_shape();
-    if(output_shape.rank().is_dynamic() || input_shape_a.rank().is_dynamic() || input_shape_b.rank().is_dynamic()) {
-        return false;
-    }
```

Comment:
Why? We cannot work with dynamic ranks: the next call to `.rank().get_length()` will throw in case if the rank is really dynamic.

Reply:
Oops, I was testing around this part but it shouldn't be changed.
Reverted.